hack/e2e.sh

```diff
@@ -2,11 +2,9 @@
 
 set -euo pipefail
 
-GOPATH="$(mktemp -d)"
-export GOPATH
+unset GOFLAGS
```

Comment:
@spangenberg I'm still confusing why we need `unset GOFLAGS`, can you elaborate?

Reply:
`hack/e2e.sh` is executing the `Makefile` in [github.com/openshift/cluster-api-actuator-pkg](https://github.com/openshift/cluster-api-actuator-pkg) which does `GOFLAGS += -mod=vendor` just like this repos `Makefile` does.
This would result in `GOFLAGS=-mod=vendor-mod=vendor` which gets passed to each `go build` command which is wrong. So I unset it here because it get's se again by the other `Makefile`.

Reply:
thanks! is there a reason to concat rather than just set?

Reply:
In case tooling sets other commands with GOFLAGS, that would overwrite it.
Found that patters in other projects too.

Reply:
but this is exactly that case and it breaks because it wrongly concat with no spaces, right?